src/store/hooks.ts

```diff
@@ -26,11 +38,24 @@ export const useAssetsSelector = (id: string) =>
     state.undoReducer.present.assets.find((item) => item.id === id)
   );
 
+export const useNewAssetsIndex = () => {
+  useAppSelector((state) => state.undoReducer.present.assets.length);
+};
+
+export const useAllocationSelector = (id: string) =>
+  useAppSelector((state) =>
+    state.undoReducer.present.allocation.find((item) => item.id === id)
+  );
+
+export const useNewAllocationIndex = () => {
+  useAppSelector((state) => state.undoReducer.present.allocation.length);
+};
+
```

Comment:
These are not really hooks they are just functions that return app selector, so i'd rather call them 
selectNewAllocationIndex for example and move it to slice. You can revise all of the functions here because they are no custom hooks. Hook interacts with life cycle of react